Sources/Sentry/SentryFileManager.m

```diff
@@ -205,6 +223,14 @@ + (BOOL)createDirectoryAtPath:(NSString *)path withError:(NSError **)error {
                                         error:error];
 }
 
+#pragma mark private methods
+
+- (void)createDirectoryIfNotExists:(NSString *)path didFailWithError:(NSError **)error {
+    if (![[NSFileManager defaultManager] fileExistsAtPath:path]) {
```

Comment:
Since these I/O APIs are inherently rancey, often there's an API to create if it doesn't exist or do nothing (with a single I/O call).

I looked it up and the current method cannot be used if the directory already exist. But also, this method was marked obsolete.

This is the replacement: [createDirectoryAtURL](https://developer.apple.com/documentation/foundation/nsfilemanager/1415371-createdirectoryaturl?language=objc)

This one seems to be OK with a call regardless if the dir exists so _worst_ case we're doing I/O once and no race using 2 calls.

Maybe we give this new method a try?

Min versions supported are:

```
    iOS 5.0+
    macOS 10.7+
    Mac Catalyst 13.0+
    tvOS 9.0+
    watchOS 2.0+
```

@HazAT where do we list our min support?

Reply:
Very good point. I just refactored it and didn't check the implementation. According to our official [docs](https://docs.sentry.io/platforms/cocoa/).

Reply:
Actually you can use the existing method `[createDirectorAtPath](https://developer.apple.com/documentation/foundation/nsfilemanager/1407884-createdirectoryatpath?language=objc)`. No need to check if the directory exists.

Reply:
I must have looked at the wrong method then since your link isn't marked as `obsolete`.
Looks good then. If we don't need the `if directExists` life is good